### PR TITLE
Skill - LiveToServe

### PR DESCRIPTION
**Pull request by JesterWizard**
Heal the user by the amount they heal the target by. 

This also accounts for healing amounts below the staff might, so if the target only heals 2 HP, the user only heals 2 HP.

I couldn't get the animation to visibly show both sides healing at the same time however. That's on my TODO list.

### CODE REVIEW

File: Wizardry/Common/ItemSys/Source/StaffHeal.c [hunk 1]
```diff
@@ -73,6 +73,15 @@ void ExecStandardHeal(ProcPtr proc)
     amount = HealAmountGetter(amount, unit_act, unit_tar);
 #endif
 
+
+#if defined(SID_LiveToServe) && (COMMON_SKILL_VALID(SID_LiveToServe))
+    if (SkillTester(unit_act, SID_LiveToServe))
+    {
+        AddUnitHp(unit_act, -(gBattleTarget.unit.curHP - GetUnitCurrentHp(unit_tar)));
```
Comment (MokhaLeee) [inline Wizardry/Common/ItemSys/Source/StaffHeal.c:80]: apparently the calculation is not consistent to the skill desc

**Comment by MokhaLeee** [comment]
Meanwhile, it is not proper to set the selfheal just in standard-heal. It is better to set in post-action. But apparently it needs more effort to add lib function to judge details of the character's action. I think you can submit the skill infos first, I'll make on it later

File: Wizardry/Common/ItemSys/Source/StaffHeal.c (hunk 1 above)
Comment (JesterWizard) [reply Wizardry/Common/ItemSys/Source/StaffHeal.c:80]: True, it's missing the skill holder's MAG value. Good spot. Not sure how I missed that.

**Comment by JesterWizard** [comment]
> Meanwhile, it is not proper to set the selfheal just in standard-heal. It is better to set in post-action. But apparently it needs more effort to add lib function to judge details of the character's action. I think you can submit the skill infos first, I'll make on it later

Understood. I have removed the skill hack.

**Pull request merged**